File: Lourd/PolyPaint/PolyPaint/Templates/ShapeStyle.cs
namespace PolyPaint.Templates
{
    public class ShapeStyle
    {
        public Coordinates coordinates { get; set; }
        public double width { get; set; }
        public double height { get; set; }
        public double rotation { get; set; }
        public string borderColor { get; set; }
        public int borderStyle { get; set; }
        public string backgroundColor { get; set; }
        
        public ShapeStyle()
        {
        }

        public ShapeStyle(Coordinates coordinates, double width, double height, double rotation, string borderColor, int borderStyle, string backgroundColor)
        {
            this.coordinates = coordinates;
            this.width = width;
            this.height = height;
            this.rotation = rotation;
            this.borderColor = borderColor;
            this.borderStyle = borderStyle;
            this.backgroundColor = backgroundColor;
        }

        public ShapeStyle Clone()
        {
            return new ShapeStyle(new Coordinates(coordinates.x, coordinates.y), width, height, rotation, borderColor, borderStyle, backgroundColor);
        }
    }
}

File: Leger/PolyPaint/app/src/main/java/com/polypaint/polypaint/Model/ShapeStyle.kt
package com.polypaint.polypaint.Model
import java.io.Serializable


class ShapeStyle (var coordinates: Coordinates, var width: Double, var height: Double, var rotation: Double, var borderColor: String, var borderStyle: Int, var backgroundColor: String): Serializable{

    fun copy(coordinates: Coordinates = this.coordinates.copy(),
             width: Double = this.width,
             height: Double = this.height,
             rotation: Double = this.rotation,
             borderColor: String = this.borderColor,
             borderStyle: Int = this.borderStyle,
             backgroundColor: String = this.backgroundColor) = ShapeStyle(coordinates,width, height, rotation,borderColor,borderStyle,backgroundColor)
}